src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,20 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::Dict{T, S}, gd) where {T, S}
+    t = ntuple(length(gd.cols)) do i 
+        key[T(_names(gd)[i])]
+    end
+end
+
+function Base.to_index(gd::GroupedDataFrame, key::Dict{T, S}) where {T<:Union{Symbol, <:AbstractString}, S}
+    if length(key) != length(gd.cols) || any(Symbol(n) ∉ _names(gd) for n in keys(key))
```

Comment:
The second test was incorrect as commented above
```suggestion
    if length(key) != length(gd.cols)
```
Also I think we do not need it as if the key is not found we will get an error trying to retrieve it in `_dict_to_tuple`. Finally I would move the `length`-check to `_dict_to_tuple` to avoid code duplication in `haskey`.